graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -388,8 +602,161 @@ def enter_field_definition(
             new_field_name = self.renamings.get(field_name, field_name)  # Default use original
             if new_field_name == field_name:
                 return IDLE
+            if new_field_name is None:
+                # Suppress the type
+                return REMOVE
             else:  # Make copy of node with the changed name, return the copy
                 field_node_with_new_name = get_copy_of_node_with_new_name(node, new_field_name)
                 return field_node_with_new_name
 
         return IDLE
+
+
+class CascadingSuppressionCheckVisitor(Visitor):
```

Comment:
Perhaps as a class docstring, explain how this visitor is intended to be used. Its API is currently undocumented — it seems it's something like "instantiate it, use it in a `visit()` call, and then look at some magic attributes".

Not only is the API not documented, but the "output" attributes aren't even defined and typed at class level: https://mypy.readthedocs.io/en/stable/class_basics.html#instance-and-class-attributes

Reply:
I've added a class docstring here and with the other visitor.

For typing at the class level vs. in-line, I believe we started a discussion a few weeks ago but I don't think we actually came to a full conclusion on it. At the time the concern was 

"How do we make `self.foo: str = "foo"` work but not `bar.foo: str = "foo"` for some random bar?"

although it turns out `mypy` does catch situations where these in-line type-hints happen in both cases where something gets type-hinted outside of the class.